dag/dag.go

```diff
@@ -285,22 +313,34 @@ func (g *AcyclicGraph) reverseDepthFirstWalk(start []Vertex, cb WalkFunc) error
 		frontier = frontier[:n-1]
 
 		// Check if we've seen this already and return...
-		if _, ok := seen[current]; ok {
+		if _, ok := seen[current.Vertex]; ok {
 			continue
 		}
-		seen[current] = struct{}{}
+		seen[current.Vertex] = struct{}{}
 
 		// Visit the current node
-		if err := cb(current); err != nil {
+		if err := f(current.Vertex, current.Depth); err != nil {
 			return err
 		}
 
-		// Visit targets of this in reverse order.
-		targets := g.UpEdges(current).List()
-		for i := len(targets) - 1; i >= 0; i-- {
-			frontier = append(frontier, targets[i].(Vertex))
+		// Visit targets of this in a consistent order.
+		targets := AsVertexList(g.UpEdges(current.Vertex))
+		sort.Sort(ByVertexName(targets))
+		for _, t := range targets {
+			frontier = append(frontier, &vertexAtDepth{
+				Vertex: t,
+				Depth:  current.Depth + 1,
+			})
 		}
 	}
 
 	return nil
 }
+
+type ByVertexName []Vertex
```

Comment:
If we're going to expose this as a public type, let's make it `SortVertexName`. That way in API docs it'll be slightly easier. Also, it needs a doc string here. 

Or, you can make it private. I don't think either way matters.